Fix test on Open Protocol Parser

ferm10n opened the pull request:
closes https://github.com/st-one-io/node-open-protocol/issues/44

joshpruett reviewed (changes requested):
Is the spirit of the test simply that an error occurs, or specifically that the error is _emitted_?

I think the test should be renamed so it's clearer which it's expecting to validate, a `throw` or an `emit`. Or since we didn't write the test and don't really know, maybe a comment stating why the previous test logic was changed for future reference.

Commit:
test: open protocol parser input is not a buffer

this unit test was failing because the error is emitted synchronously when calling `parser.write`
It's functionality provided by nodejs internals, so it's a little weird to test for this in the first place
